docs/source/models.rst

```diff
@@ -1122,6 +1122,194 @@ in native spaCy format.
+    convert(dev_df, "corpora.dev")
+
+
+To set up the training job, we first need to generate a configuration file as described in the `Spacy Documentation <https://spacy.io/usage/training#config>` _
+For simplicity, we will only use a ``TextCategorizer`` in the pipeline.
+
+.. code-block:: console
+
+    python -m spacy init config --pipeline textcat --lang en mlflow-textcat.cfg
+
+Change the default train and dev paths in the config file
+
+.. code-block:: text
+
+    [paths]
+    train = null
+    dev = null
+
+to the current directory:
+
+.. code-block:: text
+
+    [paths]
+    train = "."
+    dev = "."
+
+
+In ``Spacy``, the training loop is defined internally in Spacy's code. Spacy provides a "logging" extension point where
+we can use ``mlflow``. To do this,
+
+* We have to define a function to write metrics / model input to ``mlfow``
+* Register it as a logger in ``Spacy``'s component registry
+* Change the default console logger in the ``Spacy``'s configuration file (``mlflow-textcat.cfg``)
+
+.. code-block:: python
+
+    from typing import IO, Callable, Tuple, Dict, Any, Optional
+    import spacy
+    from spacy import Language
+    import mlflow
+
+
+    @spacy.registry.loggers("mlflow_logger.v1")
+    def mlflow_logger():
+        """Returns a function, ``setup_logger`` that returns two functions:
+
+        * ``log_step`` is called internally by Spacy for every evaluation step. We can log the intermediate train and
+        validation scores to the mlflow tracking server here.
+        * ``finalize``: is called internally by Spacy after training is complete. We can log the model artifact to the
+        mlflow tracking server here.
+        """
+
+        def setup_logger(
+            nlp: Language,
+            stdout: IO = sys.stdout,
+            stderr: IO = sys.stderr,
+        ) -> Tuple[Callable, Callable]:
+            def log_step(info: Optional[Dict[str, Any]]):
+                if info:
+                    step = info["step"]
+                    score = info["score"]
+                    metrics = {}
+
+                    for pipe_name in nlp.pipe_names:
+                        loss = info["losses"][pipe_name]
+                        metrics[f"{pipe_name}_loss"] = loss
+                        metrics[f"{pipe_name}_score"] = score
+                    mlflow.log_metrics(metrics, step=step)
+
+            def finalize():
+                uri = mlflow.spacy.log_model(nlp, "mlflow_textcat_example")
+                mlflow.end_run()
+
+            return log_step, finalize
+
+        return setup_logger
+
+
+Check the `spacy-loggers library <https://pypi.org/project/spacy-loggers/>` _ for a more complete implementation.
+
+Point to our mlflow logger in ``Spacy`` configuration file:
+
+.. code-block:: text
+
+    [training.logger]
+    @loggers = "spacy.ConsoleLogger.v1"
+    progress_bar = false
+
+section of the config file to
+
+.. code-block:: text
+
+    [training.logger]
+    @loggers = "mlflow_logger.v1"
```

Comment:
Can we apply the same fix as https://github.com/mlflow/mlflow/pull/7753/files#r1092761717 here?

Reply:
Fixed.